### PR TITLE
fix(producer): use a delayed reaction to node unhealthy status to pause block production

### PR DESCRIPTION
**Pull request by danwt**
# PR Standards

## Opening a pull request should be able to meet the following requirements

---

Close https://github.com/dymensionxyz/dymint/issues/741

<-- Briefly describe the content of this pull request -->

For Author:

- [ ]  Targeted PR against correct branch
- [ ]  included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Linked to Github issue with discussion and accepted design
- [ ]  Targets only one github issue
- [ ]  Wrote unit and integration tests
- [ ]  All CI checks have passed
- [ ]  Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code)

---

For Reviewer:

- [ ]  confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ]  Reviewers assigned
- [ ]  confirmed all author checklist items have been addressed

---

After reviewer approval:

- [ ]  In case targets main branch, PR should be squashed and merged.
- [ ]  In case PR targets a release branch, PR should be rebased.

### CODE REVIEW

**Review by zale144 — DISMISSED** [review]
Very elegant work

File: block/produce.go [hunk 1]
```diff
@@ -66,12 +66,15 @@ func (m *Manager) ProduceBlockLoop(ctx context.Context) {
 				continue
 			}
 			resetEmptyBlocksTimer()
-		case shouldProduceBlocks := <-m.shouldProduceBlocksCh:
+		case shouldProduceBlocks := <-m.nodeHealthErrorHandler.shouldProduceBlocksCh:
+			didPause := shouldProduceBlocks == false
 			for !shouldProduceBlocks {
 				m.logger.Info("block production paused - awaiting positive continuation signal")
```
[the remark this answers is not in the document]
Comment (danwt) [reply block/produce.go:72]: It does, but there's nothing else going on when things are paused, so it's kind of fine IMO, even helpful

File: block/produce.go [hunk 2]
```diff
@@ -39,6 +39,8 @@ func (m *Manager) ProduceBlockLoop(ctx context.Context) {
 	}
 
 	for {
+		m.logger.Debug("PRODUCE LOOP SANITY")
```
[the remark this answers is not in the document]
Comment (danwt) [reply block/produce.go]: yeah sorry this was only temporary XD

**Comment by danwt** [comment]
I suppose we're putting this on hold for now, due to merging a smaller change instead (retry inside celestia)

**Comment by danwt** [comment]
No point keeping this open

**Pull request closed without merging**